### PR TITLE
Run rubocop specs on CI.

### PR DESCRIPTION
**Pull request by iliabylich**

### CODE REVIEW

**Comment by iliabylich** [comment]
/cc @bbatsov Could you take a look please? Rubocop's test suite fails with 2 errors:
<details>
<summary>Log</summary>
<p>

```

==> Summary (2 workers in 43.6930s)
    [ 1]                        14219 examples, 0 failures, 5 pending       255 suites in 42.3432s      (pid 6062 exit 0 )
    [ 2]                          8529 examples, 1 failure, 2 pending       268 suites in 43.6801s      (pid 6065 exit 1 )
==> Failures
  1) RuboCop::CLI --only when one cop is given accepts cop names from plugins
     Got 2 failures from failure aggregation block.
     # ./spec/support/cli_spec_behavior.rb:27:in `block (2 levels) in <top (required)>'
     # ./lib/rubocop/rspec/shared_contexts.rb:27:in `block (4 levels) in <top (required)>'
     # ./lib/rubocop/rspec/shared_contexts.rb:26:in `chdir'
     # ./lib/rubocop/rspec/shared_contexts.rb:26:in `block (3 levels) in <top (required)>'
     # ./lib/rubocop/rspec/shared_contexts.rb:8:in `block (2 levels) in <top (required)>'
     # /home/travis/.rvm/gems/ruby-2.5.0/gems/test-queue-0.4.2/lib/test_queue/runner/rspec3.rb:37:in `block (3 levels) in run_specs'
     # /home/travis/.rvm/gems/ruby-2.5.0/gems/test-queue-0.4.2/lib/test_queue/iterator.rb:57:in `block in each'
     # /home/travis/.rvm/gems/ruby-2.5.0/gems/test-queue-0.4.2/lib/test_queue/runner.rb:406:in `around_filter'
     # /home/travis/.rvm/gems/ruby-2.5.0/gems/test-queue-0.4.2/lib/test_queue/iterator.rb:57:in `call'
     # /home/travis/.rvm/gems/ruby-2.5.0/gems/test-queue-0.4.2/lib/test_queue/iterator.rb:57:in `each'
     # /home/travis/.rvm/gems/ruby-2.5.0/gems/test-queue-0.4.2/lib/test_queue/runner/rspec3.rb:26:in `map'
     # /home/travis/.rvm/gems/ruby-2.5.0/gems/test-queue-0.4.2/lib/test_queue/runner/rspec3.rb:26:in `block (2 levels) in run_specs'
     # /home/travis/.rvm/gems/ruby-2.5.0/gems/test-queue-0.4.2/lib/test_queue/runner/rspec3.rb:25:in `block in run_specs'
     # /home/travis/.rvm/gems/ruby-2.5.0/gems/test-queue-0.4.2/lib/test_queue/runner/rspec3.rb:24:in `run_specs'
     # tasks/spec_runner.rake:62:in `run_worker'
     # /home/travis/.rvm/gems/ruby-2.5.0/gems/test-queue-0.4.2/lib/test_queue/runner.rb:301:in `block (2 levels) in spawn_workers'
     # /home/travis/.rvm/gems/ruby-2.5.0/gems/test-queue-0.4.2/lib/test_queue/runner.rb:296:in `fork'
     # /home/travis/.rvm/gems/ruby-2.5.0/gems/test-queue-0.4.2/lib/test_queue/runner.rb:296:in `block in spawn_workers'
     # /home/travis/.rvm/gems/ruby-2.5.0/gems/test-queue-0.4.2/lib/test_queue/runner.rb:293:in `times'
     # /home/travis/.rvm/gems/ruby-2.5.0/gems/test-queue-0.4.2/lib/test_queue/runner.rb:293:in `spawn_workers'
     # /home/travis/.rvm/gems/ruby-2.5.0/gems/test-queue-0.4.2/lib/test_queue/runner.rb:238:in `execute_internal'
     # /home/travis/.rvm/gems/ruby-2.5.0/gems/test-queue-0.4.2/lib/test_queue/runner.rb:129:in `execute'
     # tasks/spec_runner.rake:26:in `block in run_specs'
     # tasks/spec_runner.rake:38:in `with_encoding'
     # tasks/spec_runner.rake:24:in `run_specs'
     # tasks/spec_runner.rake:107:in `block in <top (required)>'
     # /home/travis/.rvm/gems/ruby-2.5.0/gems/rake-12.3.1/lib/rake/task.rb:271:in `block in execute'
     # /home/travis/.rvm/gems/ruby-2.5.0/gems/rake-12.3.1/lib/rake/task.rb:271:in `each'
     # /home/travis/.rvm/gems/ruby-2.5.0/gems/rake-12.3.1/lib/rake/task.rb:271:in `execute'
     # /home/travis/.rvm/gems/ruby-2.5.0/gems/rake-12.3.1/lib/rake/task.rb:213:in `block in invoke_with_call_chain'
     # /home/travis/.rvm/gems/ruby-2.5.0/gems/rake-12.3.1/lib/rake/task.rb:193:in `invoke_with_call_chain'
     # /home/travis/.rvm/gems/ruby-2.5.0/gems/rake-12.3.1/lib/rake/task.rb:182:in `invoke'
     # /home/travis/.rvm/gems/ruby-2.5.0/gems/rake-12.3.1/lib/rake/application.rb:160:in `invoke_task'
     # /home/travis/.rvm/gems/ruby-2.5.0/gems/rake-12.3.1/lib/rake/application.rb:116:in `block (2 levels) in top_level'
     # /home/travis/.rvm/gems/ruby-2.5.0/gems/rake-12.3.1/lib/rake/application.rb:116:in `each'
     # /home/travis/.rvm/gems/ruby-2.5.0/gems/rake-12.3.1/lib/rake/application.rb:116:in `block in top_level'
     # /home/travis/.rvm/gems/ruby-2.5.0/gems/rake-12.3.1/lib/rake/application.rb:125:in `run_with_threads'
     # /home/travis/.rvm/gems/ruby-2.5.0/gems/rake-12.3.1/lib/rake/application.rb:110:in `top_level'
     # /home/travis/.rvm/gems/ruby-2.5.0/gems/rake-12.3.1/lib/rake/application.rb:83:in `block in run'
     # /home/travis/.rvm/gems/ruby-2.5.0/gems/rake-12.3.1/lib/rake/application.rb:186:in `standard_exception_handling'
     # /home/travis/.rvm/gems/ruby-2.5.0/gems/rake-12.3.1/lib/rake/application.rb:80:in `run'
     # /home/travis/.rvm/gems/ruby-2.5.0/gems/rake-12.3.1/exe/rake:27:in `<top (required)>'
     1.1) Failure/Error: expect($CHILD_STATUS.success?).to be_truthy
            expected: truthy value
                 got: false
          # ./spec/rubocop/cli/cli_options_spec.rb:210:in `block (4 levels) in <top (required)>'
     1.2) Failure/Error:
                    expect(output)
                      .to eq(<<-RESULT.strip_indent)
                        Inspecting 2 files
                        ..
            
                        2 files inspected, no offenses detected
                      RESULT
            expected: "Inspecting 2 files\n..\n\n2 files inspected, no offenses detected\n"
                 got: ""
            (compared using ==)
            Diff:
            @@ -1,5 +1 @@
            -Inspecting 2 files
            -..
            -
            -2 files inspected, no offenses detected
          # ./spec/rubocop/cli/cli_options_spec.rb:211:in `block (4 levels) in <top (required)>'
```

</p>
</details>

Full log is available here - https://travis-ci.org/whitequark/parser/jobs/446760917

I also have the same error locally when I run it with `rake spec`. If I run only this specific spec it passes. Any ideas?

**Comment by bbatsov** [comment]
Hmm, that's weird - your code looks good to me. I'm on the road right now, but I'll take a look into this when I can. Let's also ping @koic, @Drenmi and @jonas054 and see if they can beat me to it! :-)

**Comment by Drenmi** [comment]
Just got back from RubyConfMY. Might have a chance to check it tomorrow. 🙂

**Comment by iliabylich** [comment]
@bbatsov @Drenmi any updates? Should I just remove this file? I don't think that CLI-related specs may reveal any bugs in parser.

**Comment by bbatsov** [comment]
@iliabylich Sorry for dropping the ball on this - was on a very long business trip and had no time for this.

Can you first check if the problem still exists with the most recent code?

File: ci/run_rubocop_specs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+#!/usr/bin/env bash
```
Comment (segiddins) [inline ci/run_rubocop_specs]: should this file have `set -eux`, so commands aren't allowed to fail?

Comment (iliabylich) [reply ci/run_rubocop_specs]: Thanks, fixed

**Comment by iliabylich** [comment]
@bbatsov Now it's my turn to apologize for such a late response. I've just pushed an update to the script and got even more errors - https://travis-ci.org/whitequark/parser/jobs/472343506. Any thoughts on what may cause these errors? Some of them are related to Rubocop's CLI (we definitely can ignore them) and also none of them are caused by parsing errors.

@whitequark Can we merge it as is and fix it later?

**Comment by whitequark** [comment]
> @whitequark Can we merge it as is and fix it later?

Can we break master for unknown amount of time? I'd say no.

**Comment by whitequark** [comment]
Ah, it's in allowed failures. Then yes.

**Comment by iliabylich** [comment]
@bbatsov @Drenmi  Maybe there are some groups of specs in rubocop that could somehow guarantee that Parser works as expected? If so could you point me to them please?

**Pull request merged**